src/ui/public/stringify/types/duration.js

```diff
@@ -0,0 +1,101 @@
+import 'ui/stringify/editors/duration.less';
+import _ from 'lodash';
+import moment from 'moment';
+import IndexPatternsFieldFormatProvider from 'ui/index_patterns/_field_format/FieldFormat';
+import durationTemplate from 'ui/stringify/editors/duration.html';
+
+export default function DurationFormatProvider(Private) {
+  const ratioToSeconds = {
+    picoseconds: 0.000000000001,
+    nanoseconds: 0.000000001,
+    microseconds: 0.000001
+  };
+  const FieldFormat = Private(IndexPatternsFieldFormatProvider);
+  const HUMAN_FRIENDLY = 'humanize';
+  const DEFAULT_OUTPUT_PRECISION = 2;
+  const DEFAULT_INPUT_FORMAT = { text: 'Seconds', kind: 'seconds' };
+  const inputFormats = [
+    { text: 'Picoseconds', kind: 'picoseconds' },
+    { text: 'Nanoseconds', kind: 'nanoseconds' },
+    { text: 'Microseconds', kind: 'microseconds' },
+    { text: 'Milliseconds', kind: 'milliseconds' },
+    DEFAULT_INPUT_FORMAT,
+    { text: 'Minutes', kind: 'minutes' },
+    { text: 'Hours', kind: 'hours' },
+    { text: 'Days', kind: 'days' },
+    { text: 'Weeks', kind: 'weeks' },
+    { text: 'Months', kind: 'months' },
+    { text: 'Years', kind: 'years' }
+  ];
+  const DEFAULT_OUTPUT_FORMAT = { text: 'Human Readable', method: 'humanize' };
+  const outputFormats = [
+    DEFAULT_OUTPUT_FORMAT,
+    { text: 'Milliseconds', method: 'asMilliseconds' },
+    { text: 'Seconds', method: 'asSeconds' },
+    { text: 'Minutes', method: 'asMinutes' },
+    { text: 'Hours', method: 'asHours' },
+    { text: 'Days', method: 'asDays' },
+    { text: 'Weeks', method: 'asWeeks' },
+    { text: 'Months', method: 'asMonths' },
+    { text: 'Years', method: 'asYears' }
+  ];
+
+  class Duration extends FieldFormat {
+    constructor(params) {
+      super(params);
```

Comment:
There's no need to specify a constructor if it's only calling the parent constructor.